packages/arb-token-bridge-ui/src/hooks/CCTP/useCCTP.ts

```diff
@@ -0,0 +1,199 @@
+type Contracts = {
+  tokenMessengerContractAddress: `0x${string}`
+  targetChainDomain: ChainDomain
+  targetChainId: CCTPSupportedChainId
+  usdcContractAddress: `0x${string}`
+  messengerTransmitterContractAddress: `0x${string}`
+  attestationApiUrl: string
+}
+
+const contracts: Record<CCTPSupportedChainId, Contracts> = {
+  [ChainId.Mainnet]: {
+    tokenMessengerContractAddress: '0xbd3fa81b58ba92a82136038b25adec7066af3155',
+    targetChainDomain: ChainDomain.ArbitrumOne,
+    targetChainId: ChainId.ArbitrumOne,
+    usdcContractAddress: CommonAddress.Mainnet.USDC,
+    messengerTransmitterContractAddress:
+      '0xC30362313FBBA5cf9163F0bb16a0e01f01A896ca',
+    attestationApiUrl: 'https://iris-api.circle.com'
+  },
+  [ChainId.Goerli]: {
+    tokenMessengerContractAddress: '0xd0c3da58f55358142b8d3e06c1c30c5c6114efe8',
+    targetChainDomain: ChainDomain.ArbitrumOne,
+    targetChainId: ChainId.ArbitrumGoerli,
+    usdcContractAddress: CommonAddress.Goerli.USDC,
+    messengerTransmitterContractAddress:
+      '0x109bc137cb64eab7c0b1dddd1edf341467dc2d35',
+    attestationApiUrl: 'https://iris-api-sandbox.circle.com'
+  },
+  [ChainId.ArbitrumOne]: {
+    tokenMessengerContractAddress: '0x19330d10D9Cc8751218eaf51E8885D058642E08A',
+    targetChainDomain: ChainDomain.Mainnet,
+    targetChainId: ChainId.Mainnet,
+    usdcContractAddress: CommonAddress.ArbitrumOne.USDC,
+    messengerTransmitterContractAddress:
+      '0x0a992d191deec32afe36203ad87d7d289a738f81',
+    attestationApiUrl: 'https://iris-api.circle.com'
+  },
+  [ChainId.ArbitrumGoerli]: {
+    tokenMessengerContractAddress: '0x12dcfd3fe2e9eac2859fd1ed86d2ab8c5a2f9352',
+    targetChainDomain: ChainDomain.Mainnet,
+    targetChainId: ChainId.Goerli,
+    usdcContractAddress: CommonAddress.ArbitrumGoerli.USDC,
+    messengerTransmitterContractAddress:
+      '0x26413e8157cd32011e726065a5462e97dd4d03d9',
+    attestationApiUrl: 'https://iris-api-sandbox.circle.com'
+  }
+}
+
+type AttestationResponse =
+  | {
+      attestation: `0x${string}`
+      status: 'complete'
+    }
+  | {
+      attestation: null
+      status: 'pending_confirmations'
+    }
+
+function getContracts(chainId: CCTPSupportedChainId | undefined) {
+  if (!chainId) {
+    return contracts[ChainId.Mainnet]
+  }
+  return contracts[chainId]
+}
+
+export type UseCCTPParams = {
+  sourceChainId?: CCTPSupportedChainId
+  walletAddress?: `0x${string}` | string
+}
+export function useCCTP({ sourceChainId, walletAddress }: UseCCTPParams) {
+  const {
+    tokenMessengerContractAddress,
+    targetChainDomain,
+    targetChainId,
+    attestationApiUrl,
+    usdcContractAddress,
+    messengerTransmitterContractAddress
+  } = getContracts(sourceChainId)
+
+  const destinationAddress = useMemo(() => {
+    if (!walletAddress) return
+
+    // CCTP uses 32 bytes addresses, while EVEM uses 20 bytes addresses
+    return utils.hexlify(utils.zeroPad(walletAddress, 32)) as `0x${string}`
+  }, [walletAddress])
```

Comment:
this is assuming that the sender's wallet address is the same as the destination address?

wondering how we're supporting custom destination address for EOAs